libafl/src/stages/tuneable.rs

```diff
@@ -28,51 +29,61 @@ struct TuneableMutationalStageMetadata {
 
 impl_serdeany!(TuneableMutationalStageMetadata);
 
+/// The default name of the tunenable mutational stage.
+pub const DEFAULT_TUNEABLE_MUTATIONAL_STAGE_NAME: &str = "TuneableMutationalStage";
+
 /// Set the number of iterations to be used by this mutational stage
-pub fn set_iters<S: HasMetadata>(state: &mut S, iters: u64) -> Result<(), Error> {
+pub fn set_iters<S: HasNamedMetadata>(state: &mut S, iters: u64, name: &str) -> Result<(), Error> {
     let metadata = state
-        .metadata_map_mut()
-        .get_mut::<TuneableMutationalStageMetadata>()
+        .named_metadata_map_mut()
+        .get_mut::<TuneableMutationalStageMetadata>(name)
         .ok_or_else(|| Error::illegal_state("TuneableMutationalStage not in use"));
     metadata.map(|metadata| {
         metadata.iters = Some(iters);
     })
 }
 
 /// Get the set iterations
-pub fn get_iters<S: HasMetadata>(state: &S) -> Result<Option<u64>, Error> {
+pub fn get_iters<S: HasNamedMetadata>(state: &S, name: &str) -> Result<Option<u64>, Error> {
     state
-        .metadata_map()
-        .get::<TuneableMutationalStageMetadata>()
+        .named_metadata_map()
+        .get::<TuneableMutationalStageMetadata>(name)
         .ok_or_else(|| Error::illegal_state("TuneableMutationalStage not in use"))
         .map(|metadata| metadata.iters)
 }
 
 /// Set the time for a single seed to be used by this mutational stage
-pub fn set_seed_fuzz_time<S: HasMetadata>(state: &mut S, fuzz_time: Duration) -> Result<(), Error> {
+pub fn set_seed_fuzz_time<S: HasNamedMetadata>(
+    state: &mut S,
+    fuzz_time: Duration,
+    name: &str,
+) -> Result<(), Error> {
     let metadata = state
-        .metadata_map_mut()
-        .get_mut::<TuneableMutationalStageMetadata>()
+        .named_metadata_map_mut()
+        .get_mut::<TuneableMutationalStageMetadata>(name)
         .ok_or_else(|| Error::illegal_state("TuneableMutationalStage not in use"));
     metadata.map(|metadata| {
         metadata.fuzz_time = Some(fuzz_time);
     })
 }
 
 /// Get the time for a single seed to be used by this mutational stage
-pub fn get_seed_fuzz_time<S: HasMetadata>(state: &S) -> Result<Option<Duration>, Error> {
+pub fn get_seed_fuzz_time<S: HasNamedMetadata>(
+    state: &S,
+    name: &str,
+) -> Result<Option<Duration>, Error> {
     state
-        .metadata_map()
-        .get::<TuneableMutationalStageMetadata>()
+        .named_metadata_map()
+        .get::<TuneableMutationalStageMetadata>(name)
         .ok_or_else(|| Error::illegal_state("TuneableMutationalStage not in use"))
         .map(|metadata| metadata.fuzz_time)
 }
 
 /// Reset this to a normal, randomized, stage
-pub fn reset<S: HasMetadata>(state: &mut S) -> Result<(), Error> {
+pub fn reset<S: HasNamedMetadata>(state: &mut S, name: &str) -> Result<(), Error> {
```

Comment:
Can you move the trait bounds in the where clause? That's the style for most of the code in libafl

Reply:
Sure, I'm just following the old code